### ISSUE TITLE
Eclipse EPIC Integration, Silent Failure

### ISSUE THREAD

**Issue by wbraswell** [Perl-Critic/Perl-Critic#582]
Perl::Critic will silently fail to run in some Eclipse EPIC installations.

A fix was found here:

https://sourceforge.net/p/e-p-i-c/discussion/258688/thread/46268530/#f8ec/4b16/9139

Specifically, we must add the following line to the beginning of the perlcritic executable:

push @ARGV, "-1";

I have done so on my own copy of perlcritic, as seen on lines 14 and 15:

https://github.com/wbraswell/rperl/blob/master/bin/_debugging/perlcritic

If this is an unsuitable fix, please help me figure out what other options we may have.  Thanks!

**Comment by otrosien** [comment]
@wbraswell could you file a bug in EPIC? I think that's the proper location to address the issue.

**Comment by wbraswell** [comment]
@otrosien 
A silent failure is always a bug.
There is no documentation or guidance on this, I had to randomly find a solution myself, so yes, this is DEFINITELY a bug.  Or, at the very least, a crazy & undocumented "feature".

**Comment by thaljef** [comment]
> There are over 270 outstanding Perl::Critic issues, so I'm pretty much assuming that none of them will ever get resolved, including this one.

I have a lot of mixed feelings about this, but basically, Perl::Critic is not actively maintained right now. I have moved on to other things and no one else has stepped forward to carry the torch. @wbraswell, if you like, I would be happy to give you a commit bit here and co-maint rights on CPAN.

**Comment by petdance** [comment]
@thaljef I didn't realize you were looking for a torch carrier.

**Comment by thaljef** [comment]
@petdance, I've asked the perlcritic mailing list for new leadership a couple times over the last few years, but nothing ever came of it. Admittedly, I was probably fuzzy on the matter. Are you interested in taking over here?

**Comment by petdance** [comment]
@thaljef That's definitely a possibility.  Let's talk in IMs some time.

**Comment by otrosien** [comment]
@wbraswell - here's the EPIC project on github: http://github.com/jploski/epic-ide/ but the bugtracker is still on sourceforge: https://sourceforge.net/projects/e-p-i-c/

The current EPIC version (0.7.2) has a "severity" option in Preferences > Perl > Perl::Critic, which you should set to "brutal" in order to pass this down to the perlcritic executor. Have you tried that?

**Comment by wbraswell** [comment]
@otrosien 
Yes I tried all the obvious and non-obvious stuff years ago.  That's why this is a bug.

**Comment by otrosien** [comment]
@wbraswell if it was a bug, it must have been fixed by now. It could be related to your local setup. I could try to assist here, but let's move that part of the discussion into the EPIC forum, because it does not belong here. In either case it's not Perl::Critic's fault :)

**Comment by wbraswell** [comment]
Okay I'm going to leave this ticket open for now as a reminder to revisit in the future if and when I have time.  Too many other bugs today!  :-)
